test/RewardsStreamerMP.t.sol

```diff
@@ -133,24 +133,101 @@ contract RewardsStreamerMPTest is Test {
         streamer.updateGlobalState();
     }
 
-    function _calculateBonusMP(uint256 amount, uint256 lockupTime) public view returns (uint256) {
+    function _calculeInitialMP(uint256 amount) internal pure returns (uint256) {
+        return amount;
+    }
+
+    function _calculateMaxAccruedMP(uint256 amount) internal view returns (uint256) {
+        return amount * streamer.MAX_MULTIPLIER();
+    }
+
+    function _calculateAbsoluteMaxTotalMP(uint256 amount) internal view returns (uint256) {
+        return _calculeInitialMP(amount) + _calculateBonusMP(amount, streamer.MAX_LOCKUP_PERIOD())
+            + _calculateMaxAccruedMP(amount);
+    }
+
+    function _calculateMaxTotalMP(uint256 amount, uint256 lockPeriod) internal view returns (uint256 maxTotalMaxMP) {
+        uint256 bonusMP = 0;
+        if (lockPeriod != 0) {
+            bonusMP = _calculateBonusMP(amount, lockPeriod);
+        }
+        return _calculeInitialMP(amount) + bonusMP + _calculateMaxAccruedMP(amount);
+    }
+
+    function _calculateBonusMP(uint256 amount, uint256 lockupTime) internal view returns (uint256) {
         return amount
             * (lockupTime * streamer.MAX_MULTIPLIER() * streamer.SCALE_FACTOR() / streamer.MAX_LOCKUP_PERIOD())
             / streamer.SCALE_FACTOR();
     }
 
-    function _calculeAccuredMP(uint256 totalStaked, uint256 timeDiff) public view returns (uint256) {
+    function _calculateAccuredMP(uint256 totalStaked, uint256 timeDiff) internal view returns (uint256) {
         return (timeDiff * totalStaked * streamer.MP_RATE_PER_YEAR()) / (365 days * streamer.SCALE_FACTOR());
     }
 
-    function _calculateTimeToMPLimit(uint256 amount) public view returns (uint256) {
+    function _calculateTimeToMPLimit(uint256 amount) internal view returns (uint256) {
         uint256 maxMP = amount * streamer.MAX_MULTIPLIER();
+        uint256 timeInSeconds = _calculateTimeToMP(amount, maxMP);
+        return timeInSeconds;
+    }
+
+    function _calculateTimeToMP(uint256 amount, uint256 target) internal view returns (uint256) {
```

Comment:
We should probably rename this. "calculate time to MP" doesn't make a lot of sense

Reply:
renamed to _calculateTimeToAccureMP